caffe2/python/operator_test/expand_op_test.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from caffe2.python import core, workspace
+from hypothesis import given
+
+import caffe2.python.hypothesis_test_util as hu
+import hypothesis.strategies as st
+import numpy as np
+import itertools as it
+
+class TestExpandOp(hu.HypothesisTestCase):
+    def run_expand_op_test(
+        self, op_name, X, gc, dc):
+        shape_length = np.random.randint(5)
+        shape_list = []
+        j = shape_length-1
+        i = X.ndim-1
+        while i >= 0 or j >= 0:
+            k = np.random.randint(5) + 1
+            if i >= 0 and X.shape[i] != 1:
+                if np.random.randint(2) == 0:
+                    k = 1
+                else:
+                    k = X.shape[i]
+            shape_list.insert(0, k)
+            i -= 1
+            j -= 1
+        shape = np.array(shape_list, dtype=np.int32)
+
+        op = core.CreateOperator(
+            op_name,
+            ["X", "shape"],
+            ["Y"],
+        )
+        def ref(X, shape):
+            return (np.array(X) * np.ones(shape),)
```

Comment:
I'm not sure if this is the best practice, consider using np.tile?

Reply:
I think np.broadcast_to is equals to this function?

Reply:
This will make things clear especially in the test. Use np.broadcast_to will let us know that what we are testing is to make sure the op has the same functionality of np.broadcast_to.

Reply:
@BIT-silence in fact we cannot use np.broadcast_to here, since our function is a little bit different from np.broadcast_to.

for example:

we have an array X, X.shape = [3,1]
let Y = caffe2.Expand(X, (3,1,3))
then Y.shape = [3,3,3], and that's what we want.

however, if we
let Y = np.broadcast_to(X, (3,1,3)),
then np will raise a value error notify you that the second dimension of new_shape is smaller than X.shape, and it is illegal.

Reply:
I see, thanks for the clarification.